root/static/scripts/edit/URLCleanup.js

```diff
@@ -1337,6 +1337,80 @@ const CLEANUPS: CleanupEntries = {
       };
     },
   },
+  'boomkat': {
+    match: [new RegExp('^(https?://)?(www\\.)?boomkat\\.com', 'i')],
+    restrict: [
+      {
+        artist: [
+          LINK_TYPES.downloadpurchase.artist,
+          LINK_TYPES.mailorder.artist,
+        ],
+        label: [
+          LINK_TYPES.downloadpurchase.label,
+          LINK_TYPES.mailorder.label,
+        ],
+      },
+      LINK_TYPES.downloadpurchase,
+      LINK_TYPES.mailorder,
+    ],
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)(?:www\.)?boomkat\.com\/([a-z]+)\/([^\/#?]+).*$/, 'https://boomkat.com/$1/$2');
+      return url;
+    },
+    select: function (url, sourceType) {
+      const m = /^https:\/\/boomkat\.com\/(artists|labels|products)\//.exec(url);
+      if (m) {
+        const prefix = m[1];
+        switch (prefix) {
+          case 'artists':
+            if (sourceType === 'artist') {
+              return [
+                LINK_TYPES.downloadpurchase.artist,
+                LINK_TYPES.mailorder.artist,
```

Comment:
Are things always available for download *and* mailorder? (this kind of thing is why we usually end up adding separate relationships for different stores, since in most cases you cannot know that an artist *does* have both purchase and mail order without checking further and even if they do now that might easily change... but the situation is always messy)

Reply:
Yes, this is a fair point. It depends whether the artist / label has digital or physical releases available to purchase for mail order. I've seen cases where only a physical release is available but a digital version is available on other sites.

I guess I should just remove the select function entirely?

Reply:
Yes in this case removing the `select` function will do.

Reply:
OK, I removed the `select` function and updated the tests.